### PR TITLE
docs: add example to url.ts

### LINKED ISSUE #1090: Improve JSDocs by adding examples of usages
Inspired by Viem docs, example:

https://viem.sh/docs/accounts/privateKey#usage

This is a big endeavor so it can be completed in multiple smaller PR-s (and more users can take it in parallel!)

IMPORTANT: add `@example` only to **exported** functions 

TODO utils:
- [ ] src/utils/calldata/validate 
- [ ] src/utils/calldata/formatter
- [ ] src/utils/calldata/propertyOrder
- [ ] src/utils/calldata/requestParser
- [ ] src/utils/calldata/responseParser
- [ ] src/utils/calldata/tuple
- [ ] src/utils/calldata/parser/index.ts
- [ ] src/utils/events/index.ts
- [ ] src/utils/hash/classHash.ts
- [ ] src/utils/hash/transactionHash/v3.ts

### PR DESCRIPTION
**Pull request by CollinsC1O**
## Motivation and Resolution
Closes #1090
...

### RPC version (if applicable)

...

## Usage related changes

- Change 1.
- ...

## Development related changes

- Change 1.
- ...

## Checklist:

- [x] Performed a self-review of the code
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [x] Linked the issues which this PR resolves
- [x] Documented the changes in code (API docs will be generated automatically)
- [x] Updated the tests
- [x] All tests are passing

### CODE REVIEW

File: src/utils/url.ts [hunk 1]
```diff
@@ -14,8 +14,19 @@ const protocolAndDomainRE = /^(?:\w+:)?\/\/(\S+)$/;
 const localhostDomainRE = /^localhost[:?\d]*(?:[^:?\d]\S*)?$/;
 const nonLocalhostDomainRE = /^[^\s.]+\.\S{2,}$/;
 
+/**
```
Comment (ivpavici) [inline src/utils/url.ts:17]: please remove leftover, and also please retarget the pr to `next-version` branch 🙏

Comment (CollinsC1O) [reply src/utils/url.ts:17]: ok @ivpavici  i will do so

Comment (CollinsC1O) [reply src/utils/url.ts:17]: @ivpavici please what do mean by leftover

Comment (CollinsC1O) [reply src/utils/url.ts:17]: done

**Review by ivpavici — APPROVED** [review]

**Pull request merged**